File: OnlineShop/Data/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;
using System.Runtime.Serialization;
using System.Text.Json.Serialization;

namespace OnlineShop.Models
{
    public class Product
    {
        public int Id { get; set; }
        public String Name { get; set; }
        [Column(TypeName = "decimal(19,4)")]
        public Decimal Price { get; set; }
        public int Quantity { get; set; }
        public DateTime UpdatedDate { get; set; }
        public int ProductTypeId { get; set; }
        public int CurrencyId { get; set; }
        [JsonIgnore]
        public virtual ProductType ProductType { get; set; }
        [JsonIgnore]
        public virtual Currency Currency { get; set; }
    }   
}

File: OnlineShop/ClientApp/src/app/interfaces/Product.ts
interface Product {
  id: number;
  name: string;
  price: number;
  quantity: number;
  updatedDate: string;
  productTypeId: number;
  currencyId: number;
  productType: ProductType;
  currency: Currency;
}
interface ProductType {
  id: number;
  description: string;
}
interface Currency {
  id: number;
  name: string;
  currencySymbol: string;
  isoCode: string;
}
interface ProductPriceUpdate {
  productId: number;
  productPrice: number;
}
